tiledb/utils/src/option.rs

```diff
@@ -114,6 +114,24 @@ where
     }
 }
 
+impl<T> OptionSubset for Box<T>
```

Comment:
This probably oughta be more general, e.g.
`impl<T> OptionSubset for &T where T: OptionSubset`

I just had a look at `PartialEq` and it looks like it has both this version and also the one you added.  So, up to you if you want to add it.

Reply:
Huh. I'll poke at that some more. I gave up a bit here when the first box wasn't doing exactly what I wanted and just added the `Box<[T]>` variant. Assuming your approach works I'll totes switch that out.

Reply:
I attempted using this approach and couldn't quite figure out how to make it work correctly. The `impl` you suggested gets into a recursive loop that causes a stack overflow. Reading up more, I also found this post that seems to be related which just says that it has to do with fundamental types, which Option isn't.